SnipperAPI/Controllers/SnippetsControllers.cs

```diff
@@ -26,14 +43,39 @@ public ActionResult<Snippet> CreateSnippet([FromBody] SnippetRequest snippetRequ
 
             // Returning the created snippet with Created status - use abstraction / restfulAPI
             return CreatedAtAction(nameof(GetSnippet), new { id = newSnippet.Id }, newSnippet);
+        }*/
+
+
+        [HttpPost("snippets")]
+        //[Authorize] // Requires authentication
+        public ActionResult<EncryptedSnippet> CreateSnippet([FromBody] SnippetRequest snippetRequest)
+        {
+            if (ModelState.IsValid)
+            {
+               // var currentUser = _userManager.GetUserAsync(User).Result;
+
+                // Encrypt the content before saving it
+                var encryptedContent =_encryptionService.Encrypt(snippetRequest.Content);
+
+                var newSnippet = new EncryptedSnippet(Id: Guid.NewGuid(), Title: snippetRequest.Title, Content: encryptedContent, CreatedAt: DateTime.UtcNow);
+                
+
+                Snippets.Add(newSnippet);
+
+                return CreatedAtAction(nameof(GetSnippet), new { id = newSnippet.Id }, newSnippet);
```

Comment:
Not 100% sure but cant we just return 

```suggestion
                return Ok(newSnippet);
```